src/generators/static/readmeFileGenerator.ts

```diff
@@ -5,39 +5,296 @@ import { Project } from "ts-morph";
+  projectName?: string;
+}
+
+/**
+ * Returns meta data information about the service, the package, and the client.
+ * @param clientDetails - the client details
+ * @param packageDetails - the package details
+ * @returns inferred metadata about the service, the package, and the client
+ */
+function createMetadata(
   clientDetails: ClientDetails,
-  packageDetails: PackageDetails,
-  project: Project
-) {
-  const readmeFileContents = `
-## Azure ${clientDetails.name} SDK for JavaScript
+  packageDetails: PackageDetails
+): Metadata {
+  const azureHuh = packageDetails.scopeName === "azure";
+  const repoURL = azureHuh
+    ? "https://github.com/Azure/azure-sdk-for-js"
+    : undefined;
+  const relativePackageSourcePath = clientDetails.options.azureOutputDirectory;
+  const packageSourceURL =
+    repoURL && `${repoURL}/tree/master/${relativePackageSourcePath}`;
+  const names = relativePackageSourcePath?.split("/").slice(1);
+  const packageParentDirectoryName = names?.[0];
+  const packageDirectoryName = names?.[1];
+  const clientClassName = clientDetails.name;
+  const clientPackageName = packageDetails.name;
+  const title = clientDetails.info?.title ?? clientClassName;
+  const serviceName =
+    /**
+     * It is a required convention in Azure swaggers for their titles to end with
+     * "Client".
+     */
+    title.match(/(.*) Client/)?.[1] ??
+    clientClassName.match(/(.*)Client/)?.[1] ??
+    title.match(/(.*) Service/)?.[1] ??
+    "Service";
+  return {
+    serviceName: serviceName,
+    clientPackageName: clientPackageName,
+    clientClassName: clientClassName,
+    azure: azureHuh,
+    relativePackageSourcePath: relativePackageSourcePath,
+    repoURL: repoURL,
+    packageSourceURL: packageSourceURL,
+    samplesURL: packageSourceURL && `${packageSourceURL}/samples`,
+    impressionURL: azureHuh
+      ? `https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-js%2Fsdk%2F${packageParentDirectoryName}%2F${packageDirectoryName}%2FREADME.png`
+      : undefined,
+    clientDescriptiveName: azureHuh
+      ? title.startsWith("Azure")
+        ? title
+        : `Azure ${title}`
+      : title,
+    description: clientDetails.info?.description,
+    apiRefURL: azureHuh
+      ? `https://docs.microsoft.com/javascript/api/${clientPackageName}`
+      : undefined,
+    packageNPMURL: `https://www.npmjs.com/package/${clientPackageName}`,
+    contributingGuideURL: repoURL && `${repoURL}/blob/master/CONTRIBUTING.md`,
+    projectName: azureHuh ? "Microsoft Azure SDK for Javascript" : undefined
+  };
+}
 
-This package contains an isomorphic SDK for ${clientDetails.name}.
+/**
+ * Creates the string for the "Getting Started" section in the README.md file.
+ * @param metadata metadata about the service, package, and the client.
+ * @returns the required getting started section in the readme file
+ */
+function writeGettingStarted(metadata: Metadata): string {
+  const text = `## Getting started
```

Comment:
I have a bit of a general question (for anyone, not specific to this PR) - is it common that we put all the text inline like this when generating static files? Have we explored simple templating engines (remarkable, handlebars, etc)

I'm not proposing we go all-in on super complex templates I'm just wondering if that's been explored at all just to separate content/layout from data and (for one example) avoid having to escape backticks everywhere

Reply:
We used to have template, if I recall correctly we moved to ts-morph as it allows more flexibility by crafting TS programmatically (@daviwil can correct me here). It may be interesting to do templates for static files though as they are usually not complex.